### PR TITLE
8297763: Fix missing stub code expansion before align() in shared trampolines

### PR DESCRIPTION
**Pull request by zhengxiaolinX**
This patch fixes missing stub code expansion logic before `align()` for AArch64 and RISC-V.

The `align()` at most creates 4-byte padding, so a `NativeInstruction::instruction_size` is enough.

I am considering pre-calculating the total trampoline sizes and allocating them in batches, but maybe after this one, for this is a quick fix to unblock https://github.com/openjdk/jdk/pull/11188. Please see that thread.

The `assert_alignment(pc());` added in the RISC-V part shows that RVC doesn't change the trampoline stub / static stub logic, so there is no need to adjust the trampoline size for it. [1]

Tested AArch64 hotspot tier1\~3, and 4 is still running; tested RISC-V hotspot tier1\~2, and 3\~4 are still running.

Thanks,
Xiaolin

[1] https://github.com/openjdk/jdk/blob/2deb318c9f047ec5a4b160d66a4b52f93688ec42/src/hotspot/cpu/riscv/macroAssembler_riscv.cpp#L3125-L3126

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297763](https://bugs.openjdk.org/browse/JDK-8297763): Fix missing stub code expansion before align() in shared trampolines

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11414/head:pull/11414` \
`$ git checkout pull/11414`

Update a local copy of the PR: \
`$ git checkout pull/11414` \
`$ git pull https://git.openjdk.org/jdk pull/11414/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11414`

View PR using the GUI difftool: \
`$ git pr show -t 11414`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11414.diff">https://git.openjdk.org/jdk/pull/11414.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/aarch64/codeBuffer_aarch64.cpp
Comment (RealFYang) [inline src/hotspot/cpu/aarch64/codeBuffer_aarch64.cpp:55]: Shoud we add a check for the real code alignment here and put maybe_expand_to_ensure_remaining() and masm.align(wordSize) operations under that check?

Comment (zhengxiaolinX) [reply src/hotspot/cpu/aarch64/codeBuffer_aarch64.cpp]: Thanks for the suggestion - yes, indeed.

**Comment by zhengxiaolinX** [comment]
Gentle ping - may I have a second review (the AArch64 part of the changes)? The latest change has passed hotspot tier1\~tier4 on both platforms.

**Review by RealFYang — APPROVED** [review]
Looks reasonable to me.

**Review by luhenry — APPROVED** [review]

**Comment by zhengxiaolinX** [comment]
Thanks for the reviews @RealFYang @luhenry! (The GHA failure is not related)

/integrate

**Comment by zhengxiaolinX** [comment]
Thank you, Tobias!

**Pull request closed without merging**
